rosetta-source/src/main/rosetta/legaldocumentation-csa-type.rosetta

```diff
@@ -1,8 +1,182 @@
+
+condition: one-of
+
+
+
+
+type CollateralInterestParameters: <"Represents the floating interest calculation and distribution parameters for a single currency.">
+    postingParty CounterpartyRoleEnum(0..1) <"Represents the party to which these parameters apply (the applicable party).  In other words, if the parameters are different depending on which party is posting/holding the collateral, for which party to the CSA (Party 1 or Party 2) that is posting the collateral do these parameters apply?">
+    marginType CreditSupportAgreementMarginTypeEnum(0..1) <"Specifies the type of margin for which interest is being calculated, if the parameters are different depending on type of margin (initial or variation).">
+    currency string(0..1) <"Specifies the currency for which the parameters are captured.">
+    interestCalculationParameters CollateralInterestCalculationParameters(0..1) <"Represents the basic interest calculation parameters.">
+    interestCalculationFrequency  CollateralInterestFrequency(0..1) <"Represents how often and when interest is calculated.">
+    interestHandlingParameters CollateralInterestHandlingParameters(0..1)  <"Represents the parameters describing how and when interest transfer occurs.">
+
+type CollateralInterestCalculationParameters: <"Represents parameters for calculating the amount the floating interest calculation, e.g.  for a single currency or defaults for all currencies.">
+    fixedRate number(0..1) <"Specifies the applicable fixed rate  if used.">
+    floatingRate CsaFloatingRate(0..1) <"Specifies the floating interest rate to be used.">
+    inBaseCurrency boolean(1..1) <"Specifies whether valuation transfers should be converted to base currency equivalent.">
+    compoundingType CompoundingTypeEnum(0..1) <"Specifies the type of compounding to be applied (None, Business, Calendar).">
+    compoundingBusinessCenter BusinessCenterEnum(0..*) <"Specifies the applicable business centers for compounding.">
+    dayCountFraction DayCountFractionEnum(1..1) <"Specifies the day count fraction to use for that currency.">
+    rounding Rounding(0..1) <"Specifies the rounding rules for settling in that currency.">
+    roundingFrequency RoundingFrequencyEnum(0..1) <"Specifies when/how often is rounding applied?">
+    withholdingTaxRate number(0..1) <"Specifies the withholding tax rate if a withholding tax is applicable.">
+
+    condition  InterestRate: required choice fixedRate, floatingRate
+    condition DCF:  dayCountFraction = DayCountFractionEnum->ACT_360 or dayCountFraction = DayCountFractionEnum->ACT_365_FIXED
+    condition CompoundingBC1:  if compoundingType = CompoundingTypeEnum->Business then compoundingBusinessCenter exists
+    condition CompoundingBC2:  if compoundingType <> CompoundingTypeEnum->Business then compoundingBusinessCenter is absent
+
+/*  Previous version of CsaFloatingRate, kept for review purposes; delete before publishing 
```

Comment:
All commented-out code should be deleted

Reply:
This is a comment, not code, but perhaps it's easier to delete it.